### ISSUE TITLE
HELP: Consensus jobs failed, tried 2 times, giving up.

### ISSUE THREAD

**Issue by sanvva** [marbl/canu#1111]
```
-- Starting 'cns' concurrent execution on Tue Oct  9 20:59:08 2018 with 121281.569 GB free disk space (30 processes; 11 concurrently)

    cd unitigging/5-consensus
    ./consensus.sh 1 > ./consensus.000001.out 2>&1
    ./consensus.sh 2 > ./consensus.000002.out 2>&1
    ./consensus.sh 3 > ./consensus.000003.out 2>&1
    ./consensus.sh 4 > ./consensus.000004.out 2>&1
    ./consensus.sh 5 > ./consensus.000005.out 2>&1
    ./consensus.sh 6 > ./consensus.000006.out 2>&1
    ./consensus.sh 7 > ./consensus.000007.out 2>&1
    ./consensus.sh 8 > ./consensus.000008.out 2>&1
    ./consensus.sh 9 > ./consensus.000009.out 2>&1
    ./consensus.sh 10 > ./consensus.000010.out 2>&1
    ./consensus.sh 11 > ./consensus.000011.out 2>&1
    ./consensus.sh 12 > ./consensus.000012.out 2>&1
    ./consensus.sh 13 > ./consensus.000013.out 2>&1
    ./consensus.sh 14 > ./consensus.000014.out 2>&1
    ./consensus.sh 15 > ./consensus.000015.out 2>&1
    ./consensus.sh 16 > ./consensus.000016.out 2>&1
    ./consensus.sh 17 > ./consensus.000017.out 2>&1
    ./consensus.sh 18 > ./consensus.000018.out 2>&1
    ./consensus.sh 19 > ./consensus.000019.out 2>&1
    ./consensus.sh 20 > ./consensus.000020.out 2>&1
    ./consensus.sh 21 > ./consensus.000021.out 2>&1
    ./consensus.sh 22 > ./consensus.000022.out 2>&1
    ./consensus.sh 23 > ./consensus.000023.out 2>&1
    ./consensus.sh 24 > ./consensus.000024.out 2>&1
    ./consensus.sh 25 > ./consensus.000025.out 2>&1
    ./consensus.sh 26 > ./consensus.000026.out 2>&1
    ./consensus.sh 27 > ./consensus.000027.out 2>&1
    ./consensus.sh 28 > ./consensus.000028.out 2>&1
    ./consensus.sh 29 > ./consensus.000029.out 2>&1
    ./consensus.sh 30 > ./consensus.000030.out 2>&1

-- Finished on Tue Oct  9 21:25:42 2018 (1594 seconds) with 121276.48 GB free disk space
----------------------------------------
-- Using slow alignment for consensus (iteration '1').
-- Configured 16 contig and 14 unitig consensus jobs.
--
-- Consensus jobs failed, retry.
--   job ctgcns/0001.cns FAILED.
--   job ctgcns/0002.cns FAILED.
--   job ctgcns/0003.cns FAILED.
--   job ctgcns/0004.cns FAILED.
--   job ctgcns/0005.cns FAILED.
--   job ctgcns/0006.cns FAILED.
--   job ctgcns/0007.cns FAILED.
--   job ctgcns/0008.cns FAILED.
--   job ctgcns/0009.cns FAILED.
--   job ctgcns/0010.cns FAILED.
--   job ctgcns/0011.cns FAILED.
--   job ctgcns/0012.cns FAILED.
--   job utgcns/0001.cns FAILED.
--   job utgcns/0002.cns FAILED.
--   job utgcns/0003.cns FAILED.
--   job utgcns/0004.cns FAILED.
--   job utgcns/0005.cns FAILED.
--   job utgcns/0006.cns FAILED.
--   job utgcns/0007.cns FAILED.
--   job utgcns/0008.cns FAILED.
--   job utgcns/0009.cns FAILED.
--   job utgcns/0010.cns FAILED.
--   job utgcns/0011.cns FAILED.
--   job utgcns/0012.cns FAILED.
--
--
-- Running jobs.  Second attempt out of 2.
----------------------------------------
-- Starting 'cns' concurrent execution on Tue Oct  9 21:25:47 2018 with 121276.48 GB free disk space (24 processes; 11 concurrently)

    cd unitigging/5-consensus
    ./consensus.sh 1 > ./consensus.000001.out 2>&1
    ./consensus.sh 2 > ./consensus.000002.out 2>&1
    ./consensus.sh 3 > ./consensus.000003.out 2>&1
    ./consensus.sh 4 > ./consensus.000004.out 2>&1
    ./consensus.sh 5 > ./consensus.000005.out 2>&1
    ./consensus.sh 6 > ./consensus.000006.out 2>&1
    ./consensus.sh 7 > ./consensus.000007.out 2>&1
    ./consensus.sh 8 > ./consensus.000008.out 2>&1
    ./consensus.sh 9 > ./consensus.000009.out 2>&1
    ./consensus.sh 10 > ./consensus.000010.out 2>&1
    ./consensus.sh 11 > ./consensus.000011.out 2>&1
    ./consensus.sh 12 > ./consensus.000012.out 2>&1
    ./consensus.sh 17 > ./consensus.000017.out 2>&1
    ./consensus.sh 18 > ./consensus.000018.out 2>&1
    ./consensus.sh 19 > ./consensus.000019.out 2>&1
    ./consensus.sh 20 > ./consensus.000020.out 2>&1
    ./consensus.sh 21 > ./consensus.000021.out 2>&1
    ./consensus.sh 22 > ./consensus.000022.out 2>&1
    ./consensus.sh 23 > ./consensus.000023.out 2>&1
    ./consensus.sh 24 > ./consensus.000024.out 2>&1
    ./consensus.sh 25 > ./consensus.000025.out 2>&1
    ./consensus.sh 26 > ./consensus.000026.out 2>&1
    ./consensus.sh 27 > ./consensus.000027.out 2>&1
    ./consensus.sh 28 > ./consensus.000028.out 2>&1

-- Finished on Tue Oct  9 21:25:50 2018 (3 seconds) with 121276.48 GB free disk space
----------------------------------------
-- Using slow alignment for consensus (iteration '2').
-- Configured 16 contig and 14 unitig consensus jobs.
--
-- Consensus jobs failed, tried 2 times, giving up.
--   job ctgcns/0001.cns FAILED.
--   job ctgcns/0002.cns FAILED.
--   job ctgcns/0003.cns FAILED.
--   job ctgcns/0004.cns FAILED.
--   job ctgcns/0005.cns FAILED.
--   job ctgcns/0006.cns FAILED.
--   job ctgcns/0007.cns FAILED.
--   job ctgcns/0008.cns FAILED.
--   job ctgcns/0009.cns FAILED.
--   job ctgcns/0010.cns FAILED.
--   job ctgcns/0011.cns FAILED.
--   job ctgcns/0012.cns FAILED.
--   job utgcns/0001.cns FAILED.
--   job utgcns/0002.cns FAILED.
--   job utgcns/0003.cns FAILED.
--   job utgcns/0004.cns FAILED.
--   job utgcns/0005.cns FAILED.
--   job utgcns/0006.cns FAILED.
--   job utgcns/0007.cns FAILED.
--   job utgcns/0008.cns FAILED.
--   job utgcns/0009.cns FAILED.
--   job utgcns/0010.cns FAILED.
--   job utgcns/0011.cns FAILED.
--   job utgcns/0012.cns FAILED.
--

ABORT:
ABORT: Canu 1.7
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
[baibu.canu.nohup.log](https://github.com/marbl/canu/files/2462758/baibu.canu.nohup.log)
```

**Comment by skoren** [comment]
All the jobs are failing which would imply there is an issue with an earlier step. What is the log in one of the failing jobs (unitigging/5-consensus/consensus.000001.out)? Did you previously restart this run?

**Comment by sanvva** [comment]
@skoren Yes I restarted this run using the corrected reads, but still got the same error.
unitigging/5-consensus/consensus.000001.out

$ cat consensus.000001.out
Running job 1 based on command line options.
number of threads     = 8 (command line)

-- Opening seqStore '../baibu.ctgStore/partitionedReads.seqStore' partition 1.
-- Opening tigStore '../baibu.ctgStore' version 1.
sizeof(abBead)     8
sizeof(abColumn)   40
sizeof(abAbacus)   168
sizeof(abSequence) 32
-- Computing consensus for b=0 to e=236792 with errorRate 0.0750 (max 0.4000) and minimum overlap 40

Working on tig 4 of length 40543 (28 children)
  unitig 4 detected 24 contains (7.91x, 75.97%) 4 dovetail (2.50x, 24.03%)
Index error in read 8906473 mSegm 0 mByte 31759168 mPart 1 expected BLOB, got 40 52 f9 c4 '@R��'
utgcns: stores/sqStore.C:415: void sqReadData::sqReadData_loadFromBlob(uint8*): Assertion `blob[0] == 'B'' failed.

Failed with 'Aborted'; backtrace (libbacktrace):
AS_UTL/AS_UTL_stackTrace.C::97 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
stores/sqStore.C::415 in _ZN10sqReadData23sqReadData_loadFromBlobEPh()
utgcns/libcns/abAbacus-addRead.C::145 in _ZN8abAbacus7addReadEP7sqStorejjjbPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEEPS2_IjP10sqReadDataS6_SaIS7_IS8_SE_EEE()
utgcns/libcns/unitigConsensus.C::980 in _ZN15unitigConsensus10initializeEPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS2_EEEPS0_IjP10sqReadDataS4_SaIS5_IS6_SC_EEE()
utgcns/libcns/unitigConsensus.C::753 in _ZN15unitigConsensus13generatePBDAGEcbP5tgTigPSt3mapIjP6sqReadSt4lessIjESaISt4pairIKjS4_EEEPS2_IjP10sqReadDataS6_SaIS7_IS8_SE_EEE()
utgcns/utgcns.C::581 in main()
(null)::0 in (null)()
(null)::0 in (null)()
(null)::0 in (null)()
./consensus.sh: line 79: 80896 Aborted                 $bin/utgcns -S ../baibu.${tag}Store/partitionedReads.seqStore -T ../baibu.${tag}Store 1 $jobid -O ./${tag}cns/$jobid.cns.WORKING -maxcoverage 40 -e 0.075 -pbdagcon -edlib -threads 8

**Comment by skoren** [comment]
You've got corruption in your previous step on disk, did you hit any errors or run out of quota? You can try removing the 4-unitigger, 5-consensus, asm.ctgStore, and asm.utgStore folders and re-run Canu.

**Comment by sanvva** [comment]
@skoren I have no idea whether hitting any errors or running out of quota. I removed all the files in unitigging/ and then restarted, and still got the same error. Maybe I need to contact our HPC admin. Thank you so much.

**Comment by skoren** [comment]
Then the corruption seems to be in the read database. Your original comment said you're using Canu 1.7 but the folders aren't named partitionedReads.seqStore in that version, they are partitionedReads.gkpStore. Did you change versions during your run? What does canu -version report? 

You can re-run just the assembly step using the trimmed reads (something like ```canu -p baibu -d resumeRun -assemble -pacbio-corrected <current_run>/asm.trimmedReads.fasta.gz```). Make sure you're using a release version though (1.7.1 is the latest) and not an intermediate checkout.

**Comment by sanvva** [comment]
ls -l /ds3200_1/proc/canu/Linux-amd64/bin/canu
-rwxr-xr-x 1 root root 24271 Apr 21 04:59 /ds3200_1/proc/canu/Linux-amd64/bin/canu

$ canu -version
Canu 1.7

The canu version is same all the time.

**Comment by sanvva** [comment]
@skoren No quota limit. How to check whether any error taken place in the earlies step? Thank you.

**Comment by skoren** [comment]
I don't think you're going to find an easily logged error, if it was reported up by the FS then Canu would have stopped. This is likely an FS corruption after the fact in the sequencing read store. 

I am still concerned about your Canu version/installation, how was it installed? The seq store name change (commit 01d182304a951846677a641197bb761674a5e662) was added on April 18th but 1.7 was released on Feb 27 so there is no way 1.7 can have a seqStore folder. If it wasn't installed from the releases page of our website, I suggest downloading the 1.7.1 release from there and re-running your assembly in a new folder (change the -d) with the assemble option and the trimmed reads as I suggested above.

**Comment by sanvva** [comment]
@skoren 
 ls /ds3200_1/proc/canu -l
total 968
-rw-r--r--  1 root root   3304 Apr 21 04:59 addCopyrights-BuildData.pl
-rw-r--r--  1 root root 876785 Apr 21 04:59 addCopyrights.dat
-rw-r--r--  1 root root  12772 Apr 21 04:59 addCopyrights.pl
-rw-r--r--  1 root root   1574 Apr 21 04:59 buildRelease.sh
drwxr-xr-x  3 root root     77 Apr 21 04:59 documentation
drwxr-xr-x  3 root root     29 Apr 21 04:59 kmer
drwxr-xr-x  6 root root     72 Apr 23 16:34 Linux-amd64
-rw-r--r--  1 root root  44345 Apr 21 04:59 logo.jpg
-rw-r--r--  1 root root    219 Apr 21 04:59 README.citation
-rw-r--r--  1 root root  18009 Apr 21 04:59 README.license.GPL
-rw-r--r--  1 root root   6506 Apr 21 04:59 README.licenses
-rw-r--r--  1 root root   1784 Apr 21 04:59 README.md
drwxr-xr-x 23 root root   4096 Apr 23 16:34 src

**Comment by sanvva** [comment]
@skoren 
Most likely the git clone way:

git clone https://github.com/marbl/canu.git
cd canu/src
make -j <number of threads>

**Comment by skoren** [comment]
Yeah a clone isn't a release so you've got untested code in the middle of some pretty large-scale changes. The dates you have don't match either of the release confirming this:

```
wget https://github.com/marbl/canu/archive/v1.7.1.tar.gz
tar xzf v1.7.1.tar.gz 
ls -lha canu-1.7.1/
-rw-rw-r--  1 root root 3.3K Jun 22 04:20 addCopyrights-BuildData.pl
-rw-rw-r--  1 root root 840K Jun 22 04:20 addCopyrights.dat
-rw-rw-r--  1 root root  13K Jun 22 04:20 addCopyrights.pl
-rw-rw-r--  1 root root 1.6K Jun 22 04:20 buildRelease.sh
drwxrwsr-x  3 root root  512 Jun 22 04:20 documentation
drwxrwsr-x  2 root root  512 Jun 22 04:20 .github
drwxrwsr-x  3 root root  512 Jun 22 04:20 kmer
-rw-rw-r--  1 root root  44K Jun 22 04:20 logo.jpg
-rw-rw-r--  1 root root  219 Jun 22 04:20 README.citation
-rw-rw-r--  1 root root  18K Jun 22 04:20 README.license.GPL
-rw-rw-r--  1 root root 6.4K Jun 22 04:20 README.licenses
-rw-rw-r--  1 root root 1.8K Jun 22 04:20 README.md
drwxrwsr-x 23 root root  32K Jun 22 04:20 src
```
and
```
 wget https://github.com/marbl/canu/archive/v1.7.tar.gz
tar xzf v1.7.tar.gz 
ls -lha canu-1.7
-rw-rw-r--  1 root root 3.3K Feb 27  2018 addCopyrights-BuildData.pl
-rw-rw-r--  1 root root 840K Feb 27  2018 addCopyrights.dat
-rw-rw-r--  1 root root  13K Feb 27  2018 addCopyrights.pl
-rw-rw-r--  1 root root 1.6K Feb 27  2018 buildRelease.sh
drwxrwsr-x  3 root root  512 Feb 27  2018 documentation
drwxrwsr-x  2 root root  512 Feb 27  2018 .github
drwxrwsr-x  3 root root  512 Feb 27  2018 kmer
-rw-rw-r--  1 root root  44K Feb 27  2018 logo.jpg
-rw-rw-r--  1 root root  219 Feb 27  2018 README.citation
-rw-rw-r--  1 root root  18K Feb 27  2018 README.license.GPL
-rw-rw-r--  1 root root 6.4K Feb 27  2018 README.licenses
-rw-rw-r--  1 root root 1.8K Feb 27  2018 README.md
drwxrwsr-x 23 root root  32K Feb 27  2018 src
```

but I'm not sure why it is reporting 1.7 as the version, it would not report this unless whoever built it also changed that code.

Either way the fix is the same, download the release from the releases page (you can use the pre-compiled binaries or source) and re-run the unitigging step in a new folder.

**Comment by sanvva** [comment]
@skoren Thanks very much! I'll reinstall the release version code and re-run the last step.

**Comment by skoren** [comment]
I think @brianwalenz found the issue, when you download the zip file from the repo (and not a checkout) it will report the last release version (1.7 in this case). I've fixed that in the commits now.

**Comment by sanvva** [comment]
I successfully got the final assembly using the old version canu installed fowling your suggest (canu -p baibu -d resumeRun -assemble -pacbio-corrected <current_run>/asm.trimmedReads.fasta.gz).

However, I'm confused with the results. I successfully assembled this specie (genome size =900M) two months ago using canu,  raw Pacbio reads 67Gbp （libreay size=20k）, and assemly genome size=910M, contigs N50=750kb. Then about 21Gb  raw Pacbio reads (library size=30kb) were additionally obtained to get more longer subreads. This run were using all 88Gb Pacbio reads, but the assembly is much poorer with contigs N50=560kb (assembly genome size=980M).  How to explain this?
